src/adapter/src/coord/message_handler.rs

```diff
@@ -427,6 +427,31 @@ impl<S: Append + 'static> Coordinator<S> {
     #[tracing::instrument(level = "debug", skip_all)]
     async fn message_compute_instance_status(&mut self, event: ComputeInstanceEvent) {
         event!(Level::TRACE, event = format!("{:?}", event));
+
+        let (instance_name, replica_name) = self
+            .catalog
+            .compute_instances()
+            .find(|i| i.id == event.instance_id)
+            .map(|i| {
+                let mut replica = event.replica_id.to_string();
+                for (name, id) in &i.replica_id_by_name {
+                    if *id == event.replica_id {
+                        replica = name.clone();
+                        break;
+                    }
+                }
+                (i.name.clone(), replica)
+            })
+            .unwrap_or_else(|| (event.instance_id.to_string(), event.replica_id.to_string()));
+
+        self.broadcast_notice(
+            Some(event.instance_id),
+            &format!(
+                "Cluster replica {}.{} changed status to: {:?}.",
+                instance_name, replica_name, event.status,
+            ),
+        );
```

Comment:
I didn't reproduce this, but I assume a replica failing still looks roughly like this: https://materializeinc.slack.com/archives/C02FWJ94HME/p1665089268236949

Would it be possible to get rid of the duplicate notices by remembering the last seen status and only notifying on changes?